### PR TITLE
Added Wallets Overview chart

### PR DESCRIPTION
**Pull request by wombat6**

### CODE REVIEW

**Comment by wombat6** [comment]
Apologies, I should have isolated some basic refactoring I did from the substantive changes. On the refactoring side I basically collapsed the pbuild.py and count.py modules into the plot.py module since they are all aimed at generating a plot.

File: src/cryptoadvance/specterext/stacktrack/app_config.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+
```
Comment (k9ert) [inline src/cryptoadvance/specterext/stacktrack/app_config.py]: unnecessary change

File: src/cryptoadvance/specterext/stacktrack/service.py [hunk 2]
```diff
@@ -56,13 +57,18 @@ def callback_add_wallettabs(self) -> list[dict[str, str]]:
...
-            # view_model.about_redirect=url_for("spectrum_endpoint.some_enpoint_here")
-            # but we do it small here and only replace a specific component:
-            view_model.tx_table_include  = "stacktrack/overview.jinja"
+            # view_model.header_and_summary_include = "..."
+            # view_model.balance_overview_include = "..."
+
+            # Redirect to our custom StackTrack controller so we can generate a chart.
+            view_model.wallets_overview_redirect = url_for("stacktrack_endpoint.wallets_overview")
+            # Replace the default tx table with one that includes a chart.
+            view_model.tx_table_include = "stacktrack/wallet/overview/overview_chart_and_tx_table.jinja"
```
Comment (k9ert) [inline src/cryptoadvance/specterext/stacktrack/service.py]: That should imho be deleted here. You cannot do both: Either redirect or template replacements.

File: src/cryptoadvance/specterext/stacktrack/controller.py [hunk 3]
```diff
@@ -44,19 +47,67 @@ def index():
...
+
+    wallets_overview_vm_dict = specter().service_manager.execute_ext_callbacks(
+        adjust_view_model, WalletsOverviewVm()
+    )
+    if len(wallets_overview_vm_dict) == 0:
+        wallets_overview_vm = WalletsOverviewVm()
+    elif len(wallets_overview_vm_dict.values()) == 1:
+        wallets_overview_vm = list(wallets_overview_vm_dict.values())[0]
+    else:
+        raise logger.error("Seems that we have more than one WalletsOverviewVm Extension")
```
Comment (k9ert) [inline src/cryptoadvance/specterext/stacktrack/controller.py]: Oh, i see, you copied the templates from "core" and now you need a `wallet_overview_vm` and the way you do it is simply calling this here. I think this should also work without calling the `execute_ext_callbacks` ?! Or maybe make your desired change here directly and not use that callback?!

File: src/cryptoadvance/specterext/stacktrack/controller.py [hunk 4]
```diff
@@ -44,19 +47,67 @@ def index():
...
 @stacktrack_endpoint.route("/wallet/<wallet_alias>/chart", methods=["GET"])
 @login_required
 def stacktrack_wallet_chart(wallet_alias: str) -> str:
-    span = request.args.get("span")
+    span: str = request.args.get("span")
     span = "1y" if span is None else span
-    wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
-    balance_plot: go.Figure = getattr(pbuild, f"build_plot_{span}")(wallet)
+    wallet: Wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
+    chart: go.Figure = getattr(plot, f"build_chart_{span}")(wallet.txlist())
```
Comment (k9ert) [inline src/cryptoadvance/specterext/stacktrack/controller.py]: I don't understand why you use getattr here?

File: src/cryptoadvance/specterext/stacktrack/service.py (hunk 2 above)
Comment (k9ert) [reply src/cryptoadvance/specterext/stacktrack/service.py]: I guess you did that here because you're misusing the extension-point-method. That's why i asked to not call it in your extension and maybe move line 68 directly in your extension after creating the object yourself.

File: src/cryptoadvance/specterext/stacktrack/controller.py (hunk 4 above)
Comment (wombat6) [reply src/cryptoadvance/specterext/stacktrack/controller.py]: It may be Python noobishness on my part. Basically I'm using reflection to pick out and invoke the desired `build_chart_{span}` function. I could do it with a bunch of if/elses but this seemed cleaner. Let me know if there's a better way to do this.

File: src/cryptoadvance/specterext/stacktrack/controller.py (hunk 3 above)
Comment (wombat6) [reply src/cryptoadvance/specterext/stacktrack/controller.py]: Oh, got it. Yeah, invoking the callback here is an unnecessary complication. Will fix.

File: src/cryptoadvance/specterext/stacktrack/service.py (hunk 2 above)
Comment (wombat6) [reply src/cryptoadvance/specterext/stacktrack/service.py]: Perfect, thanks.

File: src/cryptoadvance/specterext/stacktrack/controller.py (hunk 4 above)
Comment (k9ert) [reply src/cryptoadvance/specterext/stacktrack/controller.py]: Ok, can you refactor that into a function within the plot module? That way you just need reflection within the plot module and can encapsulate a bit better?

Comment (wombat6) [reply src/cryptoadvance/specterext/stacktrack/controller.py:89]: Good idea. Will do.

Comment (wombat6) [reply src/cryptoadvance/specterext/stacktrack/controller.py]: Great suggestion. Done.

**Comment by wombat6** [comment]
@k9ert Am I OK to merge?

**Review by k9ert — APPROVED** [review]

**Comment by k9ert** [comment]
Yes, LGTM! Do it.
